WordPress/Classes/Services/NotificationsService.swift

```diff
@@ -0,0 +1,223 @@
+import Foundation
+
+
+/**
+*  @class           NotificationsService
+*  @brief           This service encapsulates the Restful API related to WordPress Notifications.
+*/
+
+public class NotificationsService : LocalCoreDataService
+{
+    // MARK: - Aliases
+    public typealias Channel    = NotificationSettings.Channel
+    public typealias Stream     = NotificationSettings.Stream
+    
+    
+    /**
+    *  @details     Designated Initializer
+    *  @param       managedObjectContext    A Reference to the MOC that should be used to interact with the Core Data Stack.
+    */
+    public override init(managedObjectContext context: NSManagedObjectContext) {
+        super.init(managedObjectContext: context)
+        
+        if let restApi = AccountService(managedObjectContext: context).defaultWordPressComAccount()?.restApi {
+            remoteApi = restApi.hasCredentials() ? restApi : nil
```

Comment:
I'm hesitant to use `hasCredentials()` here as it implies that's how to check for validity. It's fine for now but I think @koke has mentioned reservations in the past. IMHO the `Account` shouldn't return an API object if the device doesn't have what it needs.